src/main/java/pl/edu/icm/pl/mxrdr/extension/dataset/tab/DatasetAnalysisTab.java

```diff
@@ -0,0 +1,175 @@
+    }
+
+    public String getEmbargoDateForDisplay(Dataset dataset) {
+        SimpleDateFormat format = new SimpleDateFormat(settingsService.getValueForKey(SettingsServiceBean.Key.DefaultDateFormat));
+        return dataset.getEmbargoDate().isEmpty() ? "" : format.format(dataset.getEmbargoDate().getOrNull());
+    }
+    
+    
+    public WorkflowExecution getWorkflowExecution(DatasetVersion datasetVersion) {
+        if (!datasetVersion.isDraft()) {
+            Optional<WorkflowExecution> result = workflowServiceBean.findLatestByTriggerTypeAndDatasetVersion(TriggerType.PostPublishDataset, datasetVersion.getDataset().getId(), datasetVersion.getVersionNumber(), datasetVersion.getMinorVersionNumber());
+            if (result.isPresent()) {
+                return result.get();
+            }
+        }
+        return null;
+    }
+    
+    public List<WorkflowArtifact> getArtifacts(DatasetVersion datasetVersion, WorkflowExecution workflowExecution) {
+        return workflowExecution != null ? workflowArfifactService.findAll(workflowExecution.getId()) : Lists.newArrayList();
+    }
+
+    public void downloadArtifact(String name, String location) {
+        try {
+            Faces.sendFile(workflowArfifactService.readAsStream(location).get().getInput(), name, true);
+        } catch (IOException e) {
+            String error = "Problem getting stream from " + location + ": " + e;
+            logger.warn(error);
+        }
+    }
+    
+    public String getAnalysisFailureReason(DatasetVersion datasetVersion, WorkflowExecution workflowExecution) {
+        if (datasetVersion.isDraft()) {
+            return "dataset.analysisTab.dataset.in.draft.message";
+        } else if (datasetVersion.getDataset().hasActiveEmbargo() && !isPermissionToViewFiles(datasetVersion.getDataset())) {
+            return "dataset.analysisTab.dataset.embargo.message";
+        } else if ((!datasetVersion.getDataset().hasActiveEmbargo() 
+                    || (datasetVersion.getDataset().hasActiveEmbargo() && isPermissionToViewFiles(datasetVersion.getDataset())))
+                    && isAnalysisInProgress(workflowExecution)) {
+            return "dataset.analysisTab.analysis.in.progress.message";
+        } else if ((!datasetVersion.getDataset().hasActiveEmbargo() 
+                || (datasetVersion.getDataset().hasActiveEmbargo() && isPermissionToViewFiles(datasetVersion.getDataset())))
+                && isAnalysisNotPerformed(workflowExecution)) {
+            return "dataset.analysisTab.analysis.not.performed.message";
+        } else if ((!datasetVersion.getDataset().hasActiveEmbargo() 
+                || (datasetVersion.getDataset().hasActiveEmbargo() && isPermissionToViewFiles(datasetVersion.getDataset())))
+                && isAnalysisFailure(workflowExecution)) {
+            return "dataset.analysisTab.analysis.failed.message";
+        }
```

Comment:
```
(!datasetVersion.getDataset().hasActiveEmbargo() 
                || (datasetVersion.getDataset().hasActiveEmbargo() && isPermissionToViewFiles(datasetVersion.getDataset())))
```
check in three above `else if` is redundant

Reply:
Done